### PR TITLE
Fix a bug preventing `cylc vip --workflow-name=foo` from working.

### PR DESCRIPTION
**Pull request by wxtim**
`--workflow-name foo` was fine.
Caused by removal of unwanted sys.argv for Cylc VIP not removing items not exactly matching items in the Command line options arguments.

This is a small fix for a bug without an issue.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] ~[Cylc-Doc](https://github.com/cylc/cylc-doc)~ not req'd
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by datamel — COMMENTED** [review]
I can't reproduce this bug, --worfklow-name is working fine for me with ` cylc vip`. Can you give me steps to follow?

**Comment by wxtim** [comment]
> I can't reproduce this bug, --worfklow-name is working fine for me (with `=` and without) with ` cylc vip`. Can you give me steps to follow?

1. Ensure that you are using a Cylc remote host (i.e. Cylc server) - this bug is not visible if you use `--host=localhost` or if you don't have `run hosts` setup in your global config

```
cylc vip ~/metomi/workflows/simplest --workflow-name=dfgaiufhn
```

**Review by datamel — COMMENTED** [review]
Thanks @wxtim. Bug now reproduced and various manual testing completed.  I think a rebase should pickup the mypy fix to let the CI run.

**Review by datamel — APPROVED** [review]
Thanks Tim, this is working well for me.

File: cylc/flow/option_parsers.py [hunk 1]
```diff
@@ -831,6 +831,9 @@ def cleanup_sysargv(
                     not in ['store_true', 'store_false']
                 ):
                     sys.argv.pop(index)
+            elif arg in [i.split('=')[0] for i in sys.argv]:
+                index = [i.split('=')[0] for i in sys.argv].index(arg)
```
Comment (oliver-sanders) [inline cylc/flow/option_parsers.py:835]: Could you compute this list outside of the nested for loop (i.e. at line 823) to avoid repeated recomputation.

**Review by oliver-sanders — COMMENTED** [review]
LGTM

**Comment by oliver-sanders** [comment]
Reproduced the bug and confirmed that this fixes it.

File: cylc/flow/option_parsers.py (hunk 1 above)
Comment (wxtim) [reply cylc/flow/option_parsers.py]: Yes.

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
Is it worth editing the changelog entry to make users aware of `-n` -> `-N` for `--no-detach`? This just caught me out.

**Pull request merged**